File: exercises/php/tests/tax/TaxCalculatorAfterFirstYearForExpensiveVehiclesTest.php
<?php

namespace McrDigital\PhpFundamentals3\Tax;

use DateTime;
use PHPUnit\Framework\TestCase;

require_once "DummyTaxCalculator.php";

class TaxCalculatorAfterFirstYearForExpensiveVehiclesTest extends TestCase
{
    private static DateTime $FIRST_OF_APRIL_2017;
    private TaxCalculator $taxCalculator;

    protected function setUp(): void
    {
        self::$FIRST_OF_APRIL_2017 = new DateTime("2017-04-01");
        $this->taxCalculator = new DummyTaxCalculator();
    }

    public function testSubsequentYearsTaxForPetrolIfOver40K(): void
    {
        $this->markTestSkipped('Ignored for now.');
        $vehicle = new Vehicle(206, FuelType::PETROL, self::$FIRST_OF_APRIL_2017, 50000);

        $this->assertEquals(450, $this->taxCalculator->calculateTax($vehicle));
    }

    public function testSubsequentYearsTaxForElectricIfOver40K(): void
    {
        $this->markTestSkipped('Ignored for now.');
        $vehicle = new Vehicle(206, FuelType::ELECTRIC, self::$FIRST_OF_APRIL_2017, 50000);

        $this->assertEquals(310, $this->taxCalculator->calculateTax($vehicle));
    }

    public function testSubsequentYearsTaxForAlternativeFuelIfOver40K(): void
    {
        $this->markTestSkipped('Ignored for now.');
        $vehicle = new Vehicle(206, FuelType::ALTERNATIVE_FUEL, self::$FIRST_OF_APRIL_2017, 50000);

        $this->assertEquals(440, $this->taxCalculator->calculateTax($vehicle));
    }
}

File: exercises/kotlin/src/test/kotlin/tax/TaxCalculatorAfterFirstYearForExpensiveVehiclesTest.kt
package tax

import kotlin.test.Ignore
import org.junit.jupiter.api.Test
import java.time.LocalDate
import java.time.Month
import kotlin.test.assertEquals

@Ignore
class TaxCalculatorAfterFirstYearForExpensiveVehiclesTest {

    private val taxCalculator: TaxCalculator = DummyTaxCalculator()
    private val FIRST_OF_APRIL_2017 = LocalDate.of(2017, Month.APRIL, 1)


    @Test
    fun subsequentYearsTaxForPetrolIfOver40K() {
        val vehicle = Vehicle(206, FuelType.PETROL, FIRST_OF_APRIL_2017, 50000)
        assertEquals(450, taxCalculator.calculateTax(vehicle))
    }

    @Test
    fun subsequentYearsTaxForElectricIfOver40K() {
        val vehicle = Vehicle(206, FuelType.ELECTRIC, FIRST_OF_APRIL_2017, 50000)
        assertEquals(310, taxCalculator.calculateTax(vehicle))
    }

    @Test
    fun subsequentYearsTaxForAlternativeFuelIfOver40K() {
        val vehicle = Vehicle(206, FuelType.ALTERNATIVE_FUEL, FIRST_OF_APRIL_2017, 50000)
        assertEquals(440, taxCalculator.calculateTax(vehicle))
    }
}